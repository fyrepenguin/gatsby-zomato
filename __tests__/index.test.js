import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer';
import { useStaticQuery } from 'gatsby';

import IndexPage from '../src/pages';

useStaticQuery.mockImplementationOnce(() => ({
  allRestaurant: {
    totalCount: 30,
    nodes: [
      {
        restaurant: {
          name: "K Foods",
          average_cost_for_two: 400,
          featured_image: "https://b.zmtcdn.com/data/pictures/chains/7/18750387/9a3be91fd9a330cdf19b118c90b7beea.jpg?output-format=webp",
          location: {
            latitude: "13.0193335601",
            longitude: "80.1426019147"
          },
          url: "https://www.zomato.com/chennai/k-foods-porur?utm_source=api_basic_user&utm_medium=api&utm_campaign=v2.1",
          user_rating: {
            "aggregate_rating": "3.8"
          },
          id: 18750387,
          photos_url: "https://www.zomato.com/chennai/k-foods-porur/photos?utm_source=api_basic_user&utm_medium=api&utm_campaign=v2.1#tabtop"
        }
      }
    ]
  }
}))

afterEach(cleanup)

describe("Index", () => {
  const data = {
    allRestaurant: {
      totalCount: 30,
      nodes: [
        {
          restaurant: {
            name: "K Foods",
            average_cost_for_two: 400,
            featured_image: "https://b.zmtcdn.com/data/pictures/chains/7/18750387/9a3be91fd9a330cdf19b118c90b7beea.jpg?output-format=webp",
            location: {
              latitude: "13.0193335601",
              longitude: "80.1426019147"
            },
            url: "https://www.zomato.com/chennai/k-foods-porur?utm_source=api_basic_user&utm_medium=api&utm_campaign=v2.1",
            user_rating: {
              "aggregate_rating": "3.8"
            },
            id: 18750387,
            photos_url: "https://www.zomato.com/chennai/k-foods-porur/photos?utm_source=api_basic_user&utm_medium=api&utm_campaign=v2.1#tabtop"
          }
        }
      ]
    }
  }
  it("IndexPage Renders correctly", () => {
    const tree = renderer.create(<IndexPage data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Layout correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('.ant-layout')).toBeInTheDocument()
  })

  it('renders Header correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('header')).toBeInTheDocument()
  })

  it('renders Header text correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('header h1')).toHaveTextContent('List of top restaurants in Chennai')
  })

  it('renders Content correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('main')).toBeInTheDocument()
  })
  it('renders Content children correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('.cards-container')).toBeInTheDocument()
  })

  it('renders Footer correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('footer')).toBeInTheDocument()
  })

  it('renders Footer text correctly', () => {
    const { container } = render(<IndexPage data={data} />)
    expect(container.querySelector('footer')).toHaveTextContent('Developed By Mourya')
  })
})