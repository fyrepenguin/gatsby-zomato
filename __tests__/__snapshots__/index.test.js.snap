// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index IndexPage Renders correctly 1`] = `
<section
  className="ant-layout"
>
  <header
    className="ant-layout-header header"
    style={
      Object {
        "height": 78,
      }
    }
  >
    <h1
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
        }
      }
    >
      List of top restaurants in Chennai
    </h1>
  </header>
  <main
    className="ant-layout-content"
    style={
      Object {
        "paddingTop": "3rem",
      }
    }
  >
    <div
      className="cards-container"
    >
      <div
        className="ant-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="K Foods"
            data-testid="card-img"
            src="https://b.zmtcdn.com/data/pictures/chains/7/18750387/9a3be91fd9a330cdf19b118c90b7beea.jpg?output-format=webp"
            style={
              Object {
                "borderRadius": "0.5rem",
                "cursor": "initial",
              }
            }
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h3
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            K Foods
          </h3>
          <div
            className="ant-typography"
            data-testid="price"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            <span
              className="ant-typography"
              style={
                Object {
                  "WebkitLineClamp": null,
                }
              }
            >
              <strong>
                Average Price: 
              </strong>
            </span>
            â‚¹ 400
          </div>
          <div
            className="ant-typography"
            data-testid="rating"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            <span
              className="ant-typography"
              style={
                Object {
                  "WebkitLineClamp": null,
                }
              }
            >
              <strong>
                User Rating: 
              </strong>
            </span>
            3.8 /5
          </div>
          <div
            className="links"
          >
            <a
              href="https://www.zomato.com/chennai/k-foods-porur?utm_source=api_basic_user&utm_medium=api&utm_campaign=v2.1"
              rel="noreferrer"
              target="_blank"
              title="Restaurant Page"
            >
              View More
            </a>
            <div
              className="ant-divider ant-divider-vertical"
              role="separator"
              style={
                Object {
                  "borderColor": "rgba(0,0,0,0.65)",
                  "height": "auto",
                }
              }
            />
            <a
              href="https://maps.google.com/?q=13.0193335601,80.1426019147"
              rel="noreferrer"
              target="_blank"
              title="location"
            >
              View Location
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="ant-layout-footer"
  >
    Developed By 
    <a
      href="https://www.github.com/fyrepenguin"
      rel="noreferrer"
      target="_blank"
    >
      Mourya
    </a>
  </footer>
</section>
`;
